crate/server/src/start_kms_server.rs

```diff
@@ -707,26 +680,88 @@ pub async fn prepare_kms_server(
             .service(get_version);
 
         app.service(default_scope)
-    })
-    // .client_disconnect_timeout(std::time::Duration::from_secs(30)) // default: 5s
-    // .tls_handshake_timeout(std::time::Duration::from_secs(18)) // default: 3s
-    // .keep_alive(std::time::Duration::from_secs(90)) // default: 5s
-    // .client_request_timeout(std::time::Duration::from_secs(90)) // default: 5s
-    // .shutdown_timeout(180); // default: 30s
-;
-    Ok(match builder {
+    });
+
+    Ok(match tls_config {
         Some(cert_auth_builder) => {
             if use_cert_auth {
+                trace!("Using Client Certificate Authentication with Rustls 0.23");
                 // Start an HTTPS server with PKCS#12 with client cert auth
                 server
                     .on_connect(extract_peer_certificate)
-                    .bind_openssl(address, cert_auth_builder)?
+                    .bind_rustls_0_23(address, cert_auth_builder)?
                     .run()
             } else {
+                trace!("Not using Client Certificate Authentication with Rustls 0.23");
                 // Start an HTTPS server with PKCS#12 but not client cert auth
-                server.bind_openssl(address, cert_auth_builder)?.run()
+                server.bind_rustls_0_23(address, cert_auth_builder)?.run()
             }
         }
         _ => server.bind(address)?.run(),
     })
 }
+
+// Client Certificate Authentication
+// Build a rustls ServerConfig supporting client cert auth
+pub(crate) fn create_rustls_server_config(server_config: &TlsParams) -> KResult<ServerConfig> {
+    trace!("Creating Rustls ServerConfig with TLS parameters: {server_config:?}");
+    // We need an initialized crypto provider to use rustls
+    initialize_aws_lc_crypto_provider();
```

Comment:
No sure what happens if you call initialize_aws_lc_crypto_provider twice (in addition to the Socket Server)